server/middleware/auth.js

```diff
@@ -0,0 +1,33 @@
+const { getUserBySessionId } = require('../lib/db');
+
+/**
+ * Authentication middleware.
+ *
+ * Sets req.user to the logged in user
+ * Sets req.user to undefined if no user is logged in
+ */
+const authMiddleware = async (req, res, next) => {
+    const { sessionId } = req.cookies;
+
+    // Check for missing parameters
+    if (sessionId === undefined || typeof sessionId !== 'string') {
+        req.user = undefined;
+        next();
+        return;
```

Comment:
Any reason why you're not just returning `next()` itself? I assume I'm missing something here 🤔